File: 프로그래머스/1/12931. 자릿수 더하기/자릿수 더하기.java
public class Solution {
    public int solution(int n) {
        int answer = 0;

        // n이 0이 될 때까지 반복하면서 자릿수를 더함
        while (n > 0) {
            answer += n % 10;  // 마지막 자릿수를 더함
            n /= 10;  // n을 10으로 나누어 자릿수를 줄임
        }

        return answer;
    }
}

File: 프로그래머스/1/12931. 자릿수 더하기/자릿수 더하기.kt
class Solution {
    fun solution(n: Int): Int {
        var answer = 0
        var num = n

        while (num > 0) {
            answer += num % 10
            num /= 10
        }

        return answer
    }
}

fun main() {
    val sol = Solution()
    println(sol.solution(123)) // 6
    println(sol.solution(987)) // 24
}
